Update `README.md` (#251)
MetaMask/api-specs: README.md

JSON-RPC API provided by MetaMask.

Previously, MetaMask dapp developers lacked a formal RPC specification. This repository contains a specification that documents which methods MetaMask supports in a way that's versionable, human-readable, and machine-readable. This spec improves the accuracy of documentation, API, and clients.

MetaMask's effectiveness as an enterprise-grade application platform depends on its ability to provide a reliable and predictable developer experience.

## Contributing

> **Note:** To make a significant change to the API, please submit a [MIP](https://github.com/MetaMask/metamask-improvement-proposals/tree/main).

1. Edit the API specification in the `openrpc.yaml` file.
2. Run `yarn install` if you haven't previously set up the repository. 
3. Run `yarn run build` to re-generate the output file: `dist/build/openrpc.json`.
4. To view the result, paste that file's contents into [the OpenRPC playground](https://playground.open-rpc.org/).

### Build Process

> The build process:
> - filters out methods that are not supported/implemented by MetaMask
> - merges wallet methods with execution API methods
> - outputs to `dist` for release
> 
> <details>
>   <summary>Build and merge summary</summary>

Reviewer: ```suggestion
```

I prefer to remove this first list and just keep the following more detailed list to reduce redundancy. We can also take it out of the dropdown in this case.

Author: OK -- I prefer the get a short version and expand to find out more approach, but it's not a hill I am keen to die on

> The build includes the script `merge-openrpc.js`, this script:
> 
> - Loads openrpc.yaml and multichain/openrpc.yaml from the local file system.
> - Fetches the Ethereum OpenRPC JSON spec from a [remote URL](https://raw.githubusercontent.com/ethereum/execution-apis/59e6a6f9947859e8bb41bc63b248aa026b0781bd/refs-openrpc.json) and filters the methods using filterExecutionAPIs.
> - Merges the local MetaMask OpenRPC specification with the Ethereum OpenRPC specification.
> - Adds tags to methods: Tags each method in the Ethereum OpenRPC specification.
> - Writes out the merged and filtered OpenRPC specifications to temporary files:
> 	- src/build/openrpc.json.
> 	- src/build/multichain-openrpc.json.
> 		
> These files are then output to the `dist` folder. 
> </details>

Reviewer: ```suggestion
The build uses the script `merge-openrpc.js`, which:

1. Loads `openrpc.yaml` and `multichain/openrpc.yaml` from the local file system.
2. Fetches the Ethereum execution API OpenRPC spec from a [remote URL](https://raw.githubusercontent.com/ethereum/execution-apis/59e6a6f9947859e8bb41bc63b248aa026b0781bd/refs-openrpc.json) and filters out methods that are not supported/implemented by MetaMask, using `filterExecutionAPIs`.
3. Merges the local MetaMask OpenRPC spec with the Ethereum execution API OpenRPC spec.
4. Tags each execution API method.
5. Writes out the merged and filtered OpenRPC spec to temporary files:
	 - `src/build/openrpc.json`
	 - `src/build/multichain-openrpc.json`
6. Outputs these files to the `dist` folder. 
```

Author: Numbers work up to 6 -- which is not a step handled by merge-openrpc AFAIK

JSON-RPC API provided by MetaMask.

> MetaMask dapp developers lack a formal RPC specification. This attempts to document which methods MetaMask supports in a way that's versionable, human- and machine-readable. It will improve the accuracy of documentation, API, and clients.

Reviewer: ```suggestion
Previously, MetaMask dapp developers lacked a formal RPC specification. This repository contains a specification that documents which methods MetaMask supports in a way that's versionable, human-readable, and machine-readable. This spec improves the accuracy of documentation, API, and clients.
```

Suggestion. `versionable, human-readable, and machine-readable` makes more sense because there are more than two items in the list.

MetaMask's effectiveness as an enterprise-grade application platform depends on its ability to provide a reliable and predictable developer experience.

## Contributing

> > Before making any significant changes to the API spec, consider whether to first submit a [MIP](https://github.com/MetaMask/metamask-improvement-proposals/tree/main) and complete the MIP [workflow](https://github.com/MetaMask/metamask-improvement-proposals/blob/main/PROCESS-GUIDE.md).

Reviewer: ```suggestion
> **Note:** To make a significant change to the API, please submit a [MIP](https://github.com/MetaMask/metamask-improvement-proposals/tree/main).
```

> Edit the `openrpc.yaml` file.
> 
> Run `yarn build` to regenerate the output file: `dist/build/openrpc.json`.
> 
> > To view the result, paste that file's contents into [the OpenRPC playground](https://playground.open-rpc.org/).

Reviewer: ```suggestion
1. Edit the API specification in the `openrpc.yaml` file.
2. Run `yarn install` if you haven't previously set up the repository. 
3. Run `yarn run build` to re-generate the output file: `dist/build/openrpc.json`.
4. To view the result, paste that file's contents into [the OpenRPC playground](https://playground.open-rpc.org/).
```

Suggestion, make into ordered list. The testing/viewing result should be a required step. Also does `yarn run build` not work for you? We should not change documented commands without explicit reason.

Author: sure, done locally

### Build Process

To build, run:

- `yarn install`, if you haven't setup the repo

> - `yarn build`

Reviewer: Remove, to reduce redundancy with contributing steps.

Author: moved/merged to Contributing

The build process will:
- filter out methods that are not supported/implemented by MetaMask
- merge wallet methods with execution API methods
- output to `dist` for release

<details>
  <summary>Build and merge summary</summary>
    <div>
	  The build includes the script `merge-openrpc.js`, this script:

		- Loads openrpc.yaml and multichain/openrpc.yaml from the local file system.
		- Fetches the Ethereum OpenRPC JSON spec from a [remote URL](https://raw.githubusercontent.com/ethereum/execution-apis/59e6a6f9947859e8bb41bc63b248aa026b0781bd/refs-openrpc.json) and filters the methods using filterExecutionAPIs.
		- Merges the local MetaMask OpenRPC specification with the Ethereum OpenRPC specification.
		- Adds tags to methods:
			> Adds a tag to each method in the Ethereum OpenRPC specification.
		- Writes out the merged and filtered OpenRPC specifications to temporary files:
			- src/build/openrpc.json.
			- src/build/multichain-openrpc.json.

	These files are then output to the `dist` folder. 
 </div>

> </details>

Reviewer: These lists should probably be combined instead of repeated. Also, the dropdown list doesn't render properly:
<img width="1411" alt="Screenshot 2024-08-12 at 3 08 42 PM" src="https://github.com/user-attachments/assets/2d9f1422-09a7-4682-b46b-d3a8971302df">

Author: weird, vscode preview fine 
![image](https://github.com/user-attachments/assets/e632cde9-7a21-4bc7-ba29-6b36a1706114)

Author: GitHub also renders this page fine: 
<img width="411" alt="image" src="https://github.com/user-attachments/assets/b31ddc16-952e-4c36-aa9d-30efefa7533a">

Author: what are you previewing with?

Reviewer: Oh, the dropdown is collapsed by default as expected. I'm noting that when you open the dropdown, most of the text appears in a code block, which I assume is not intended. This is probably due to indentation issues. I'm previewing on GH.

Author: ah -- it's an indenting issue --> Fixed
